### ISSUE TITLE
ec2_ami - snapshot not deleted when deregistering AMI

### ISSUE THREAD

**Issue by Constantin07** [ansible/ansible-modules-core#1496]
##### Issue Type:

When AMI image is de-registered and parameter "delete_snapshot: True" is set the AMI image is deleted but the snapshot is not cleaned up.
##### Ansible Version:

ansible-1.9.1-1.el7.noarch
##### Ansible Configuration:

```
   - name: Delete an AMI
     ec2_ami:
      image_id: "{{ image_id }}"
      delete_snapshot: True
      state: absent
      wait: yes
     register: ami_del_res
     tags: ami_delete
```
##### Environment:

CentOS 7. Ansible installed from EPEL repo.
##### Summary:

It seems that "delete_snapshot" parameter doesn't work as expected.
##### Steps To Reproduce:

Assuming an AMI image already exists, I run this command:

```
ansible-playbook -i hosts aws-image.yml -t ami_delete -e "image_id=ami-e3601e94" 
```

and get this result:

```
PLAY [localhost] **************************************************************

TASK: [Delete an AMI] *********************************************************
changed: [127.0.0.1]

TASK: [debug var=ami_del_res] *************************************************
ok: [127.0.0.1] => {
    "var": {
        "ami_del_res": {
            "changed": true,
            "invocation": {
                "module_args": "",
                "module_name": "ec2_ami"
            },
            "msg": "AMI deregister/delete operation complete"
        }
    }
}
```
##### Expected Results:

The snapshots should be deleted too.
##### Actual Results:

The snapshots are not deleted.

Thanks,
Constantin

**Comment by mijndert** [comment]
Any news on this?

**Comment by Constantin07** [comment]
This is already addressed in #2320

**Comment by jctanner** [comment]
Hi!

We believe recent commits (likely detailed above) should resolve this question or problem for you.

This will also be included in the next major release.

If you continue seeing any problems related to this issue, or if you have any further questions, please let us know by stopping by one of the two mailing lists, as appropriate:
- https://groups.google.com/forum/#!forum/ansible-project - for user questions, tips, and tricks
- https://groups.google.com/forum/#!forum/ansible-devel - for strategy, future planning, and questions about writing code

Because this project is very active, we're unlikely to see comments made on closed tickets, but the mailing list is a great way to ask questions, or post if you don't think this particular
issue is resolved.

Thank you!
